### DISCUSSION TITLE
Code executing in only one worker

### DISCUSSION THREAD

**Discussion by ArijitSinghEDA** [benoitc/gunicorn#3137] [Q&A]
I am spawning 4 workers in a docker with pids 1, 2, 3, 4 using gunicorn and FastAPI with uvicorn workers.

When a process is running on worker 3, it gets executed completely, but when it runs on any other worker, it stops executing mid way.

Structure of my file data:
```
main_folder
|--common_files
    |--main.py
    |--gunicorn_config.py
|--run_files
    |--running_file.py
```

`main.py`
```
app = FastAPI()

@app.post("/")
async def func():
    from run_files import running_file
    running_file.main()
    return 1
```

`gunicorn_config.py`
```
workers = 4
worker_class = 'uvicorn.workers.UvicornWorker'
bind = '127.0.0.1:8111'
timeout = 86400
graceful_timeout = 86400
```

`running_file.py`
```
def main():
    print("Hello")
    run()

def run():
    print("Yes")
```

Now, if this code runs on worker 3, I get output on the kernel
```
Hello
Yes
```

But, if it is running on any other worker, I get the output
```
Hello
```

What can be the reason for such behaviour? It wasn't happening before, it just started out of the blue.

**Comment by benoitc** [comment]
this was probably because your code was sharing some context and was locked on one process.
